metron-platform/metron-solr/src/main/java/org/apache/metron/solr/dao/SolrDao.java

```diff
@@ -168,18 +168,21 @@ public Document removeCommentFromAlert(CommentAddRemoveRequest request, Document
   }
 
   /**
-   * Builds a Solr client using the ZK hosts from the global config.
+   * Returns the SolrClient.
    * @return SolrClient
    */
   public SolrClient getSolrClient() {
-    return new CloudSolrClient.Builder().withZkHost(getZkHosts()).build();
+    if (this.client == null) {
+      throw new IllegalStateException("The SolrDao must be initialized first");
+    }
+    return this.client;
   }
 
   /**
    * Builds a Solr client using the ZK hosts specified.
    * @return SolrClient
    */
-  public SolrClient getSolrClient(List<String> zkHosts) {
+  protected SolrClient getSolrClient(List<String> zkHosts) {
     return new CloudSolrClient.Builder().withZkHost(zkHosts).build();
```

Comment:
Why would we ever want to create new clients from within a DAO class? Can we get rid of this altogether and/or get the client creation code into a proper factory? I think having `getSolrClient()` return an existing client and `getSolrClient(List<String> zkHosts)` opens us up to a relapse of this problem.

Reply:
I moved the SolrClient creation to a factory.  This should closely match the way we do it for the `ElasticsearchDao` classes.  The `SolrDao` and `SolrMetaAlertDao` are created separately in `IndexConfig` so I had to keep the `SolrDao.getSolrClient` method.  Now it only returns the current client and throws an exception if `SolrDao` hasn't been initialized.